### PR TITLE
Add AB Test for SignInGate Copy test

### PR DESCRIPTION
**Pull request by lindseydew**
## What does this change?

This adds an AB Test Switch for enabling the SignInGate Copy Test to run on dotcom rendering

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## PR for dotcom:
https://github.com/guardian/dotcom-rendering/pull/7030

Doc for the hypothesis: https://docs.google.com/document/d/1lBufse0NxTZqRNFFRbch1hcFlyv65gEyNyMHtoqKj48/edit

## Screenshots

register
take-a-moment

-->

## What is the value of this and can you measure success?

We want to discover the impact of changing the call to action message for the sign in gate that appears after a user has read three or more articles. Currently there have been complaints received relating to the messaging implying that the sign in flow is a mandatory action, when it is optional. We would validate whether we can use a more transparent message without a negative impact of sign ups. 

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: common/app/conf/switches/ABTestSwitches.scala [hunk 1]
```diff
@@ -104,4 +104,13 @@ trait ABTestSwitches {
     sellByDate = Some(LocalDate.of(2023, 2, 1)),
     exposeClientSide = true,
   )
+  Switch(
+    ABTests,
+    "ab-sign-in-gate-copy-test",
+    "Test the impact of changing the copy in the sign in gate",
+    owners = Seq(Owner.withEmail("personalisation.dev@theguardian.com")),
+    safeState = Off,
+    sellByDate = Some(LocalDate.of(2023, 2, 1)),
```
Comment (bryophyta) [inline common/app/conf/switches/ABTestSwitches.scala:113]: it looks like the `sellByDate` is set for tomorrow, is that intentional?

File: common/app/conf/switches/ABTestSwitches.scala [hunk 2]
```diff
@@ -104,4 +104,13 @@ trait ABTestSwitches {
     sellByDate = Some(LocalDate.of(2023, 2, 1)),
     exposeClientSide = true,
   )
+  Switch(
+    ABTests,
+    "ab-sign-in-gate-copy-test",
+    "Test the impact of changing the copy in the sign in gate",
+    owners = Seq(Owner.withEmail("personalisation.dev@theguardian.com")),
```
Comment (vlbee) [inline common/app/conf/switches/ABTestSwitches.scala:111]: While this email address would make logical sense it doesn't exist. There is only `personalisation@guardian.co.uk`

File: common/app/conf/switches/ABTestSwitches.scala (hunk 1 above)
Comment (vlbee) [reply common/app/conf/switches/ABTestSwitches.scala:113]: I usually set the sell by date with generous gap after the anticipated end of the test. I believe the sell by date kills the switch and therefore the test. 

We should probably raise a PR to amend the Main sign in gate switches to `sellByDate = never`

File: static/src/javascripts/projects/common/modules/experiments/tests/sign-in-gate-copy-test-variant.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	start: '2023-01-23',
+	expiry: '2025-12-01',
+	author: 'Lindsey Dew',
+	description: 'Test varying the copy in the call to action for sign in gate',
+	audience: 0.1,
+	audienceOffset: 0.0,
+	successMeasure: 'Users sign in or create a Guardian account',
+	audienceCriteria: '3rd article of the day, lower priority than consent banner, simple articles (not gallery, live etc.), not signed in, not shown after dismiss, not on help, info sections etc. Exclude iOS 9 and guardian-live-australia. Suppresses other banners, and appears over epics',
+	dataLinkNames: 'SignInGateCopyTest',
+	idealOutcome:
```
Comment (vlbee) [inline static/src/javascripts/projects/common/modules/experiments/tests/sign-in-gate-copy-test-variant.js:12]: We should ideally be able to add some firmer result numbers in ideal outcome as part of the test hypothesis once we get that info from D&I

Comment (lindseydew) [reply static/src/javascripts/projects/common/modules/experiments/tests/sign-in-gate-copy-test-variant.js:12]: Updated this from the doc: https://docs.google.com/document/d/1lBufse0NxTZqRNFFRbch1hcFlyv65gEyNyMHtoqKj48/edit

File: common/app/conf/switches/ABTestSwitches.scala (hunk 1 above)
Comment (lindseydew) [reply common/app/conf/switches/ABTestSwitches.scala]: Updated this to the end of may

File: static/src/javascripts/projects/common/modules/experiments/tests/sign-in-gate-copy-test-variant.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+export const signInGateCopyTestJan2023 = {
```
Comment (OllysCoding) [inline static/src/javascripts/projects/common/modules/experiments/tests/sign-in-gate-copy-test-variant.js:1]: Not massively familar with the client side ab testing framework, is there a reason we need to define this in Frontend if (I'm assuming) it's only used in DCR?

**Review by OllysCoding — APPROVED** [review]
+1 LGTM :)

**Review by vlbee — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/experiments/tests/sign-in-gate-copy-test-variant.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	successMeasure: 'Users sign in or create a Guardian account',
+	audienceCriteria: '3rd article of the day, lower priority than consent banner, simple articles (not gallery, live etc.), not signed in, not shown after dismiss, not on help, info sections etc. Exclude iOS 9 and guardian-live-australia. Suppresses other banners, and appears over epics',
+	dataLinkNames: 'SignInGateCopyTest',
+	idealOutcome:
+		'One variants performs at least 2% better than the control and/OR 10% better than the other variant. Neither variant performs 5% worse than the control',
+	showForSensitive: false,
+	canRun: () => true,
+	variants: [
+		{
+			id: 'quick-and-easy',
```
[the remark this answers is not in the document]
Comment (lindseydew) [reply static/src/javascripts/projects/common/modules/experiments/tests/sign-in-gate-copy-test-variant.js:18]: Added

File: common/app/conf/switches/ABTestSwitches.scala (hunk 2 above)
Comment (lindseydew) [reply common/app/conf/switches/ABTestSwitches.scala]: Updated the email address

**Pull request merged**